lib/src/git.rs

```diff
@@ -364,15 +364,19 @@ pub fn fetch(
     fetch_options.proxy_options(proxy_options);
     let callbacks = callbacks.into_git();
     fetch_options.remote_callbacks(callbacks);
-    if globs.iter().any(|g| g.contains(|c| ":^".contains(c))) {
-        return Err(GitFetchError::InvalidGlob);
-    }
-    // At this point, we are only updating Git's remote tracking branches, not the
-    // local branches.
-    let refspecs = globs
-        .iter()
-        .map(|glob| format!("+refs/heads/{glob}:refs/remotes/{remote_name}/{glob}"))
-        .collect_vec();
+    let refspecs = if let Some(globs) = branch_name_globs {
+        if globs.iter().any(|g| g.contains(|c| ":^".contains(c))) {
```

Comment:
I think that if you make this change, you need to special case `Some([])`. I'd just abort in that case, not sure whether normally or with an error.

Reply:
To be honest, I'd prefer of getting rid of the entire commit that uses `Option` here, since it only increases the number of special cases. Instead, I think documenting that `[]` means everything is good enough.

However, I don't insist on this.

Reply:
In the `Some([])` case, we could skip most of the rest of the function, probably including the import.

I think the new differentiation between `None` and `Some([])` is good, especially in a library, since it will avoid surprising behavior if the caller generates the list of globs somehow and the programmer forgot to check if the list was empty.

Reply:
Yeah, I'd just return then and there.

Reply:
We can add the "optimization" (which in reality should never trigger) later; it needs discussing whether it is worth it, and at what point it should be added (do we still want to download data at all, and check if the remote repository is still alive, etc.?).

Reply:
> **Update** On second thought, if the special case is accounted for, the `Option` might arguably be less confusing, and you already did the work of setting it up.

FWIW, never hesitate to suggest forgetting about a change I propose or doing it differently. I don't consider that because some work has been done it must be accepted or kept as-is. Only the final state must be judged on, not the quantity of work that has been put in it.